components/ModEditFacilityOrProfessionalTopbar.vue

```diff
@@ -152,42 +148,34 @@ const facilityHasUnsavedChanges = () => {
 
     /** This needs to be converted because to access the values of this object we do not need value.
     But to keep their reactivity in the store we keep them as Refs **/
-    const facilitySections = facilitiesStore.facilitySectionFields as unknown as {
-        [key: string]: string
-    }
-
-    const areThereUnsavedChanges
-        = facilityBeforeChange.nameEn !== facilitySections.nameEn
-        || facilityBeforeChange.nameJa !== facilitySections.nameJa
-        || facilityBeforeChange.contact.phone !== facilitySections.phone
-        || facilityBeforeChange.contact.website !== facilitySections.website
-        || facilityBeforeChange.contact.email !== facilitySections.email
-        || facilityBeforeChange.contact.address.postalCode !== facilitySections.postalCode
-        || facilityBeforeChange.contact.address.prefectureEn !== facilitySections.prefectureEn
-        || facilityBeforeChange.contact.address.cityEn !== facilitySections.cityEn
-        || facilityBeforeChange.contact.address.addressLine1En !== facilitySections.addressLine1En
-        || facilityBeforeChange.contact.address.addressLine2En !== facilitySections.addressLine2En
-        || facilityBeforeChange.contact.address.prefectureJa !== facilitySections.prefectureJa
-        || facilityBeforeChange.contact.address.cityJa !== facilitySections.cityJa
-        || facilityBeforeChange.contact.address.addressLine1Ja !== facilitySections.addressLine1Ja
-        || facilityBeforeChange.contact.address.addressLine2Ja !== facilitySections.addressLine2Ja
-        || facilityBeforeChange.contact.googleMapsUrl !== facilitySections.googlemapsURL
-        || facilityBeforeChange.mapLatitude.toString() !== facilitySections.mapLatitude
-        || facilityBeforeChange.mapLongitude.toString() !== facilitySections.mapLongitude
-        || JSON.stringify(facilityBeforeChange.healthcareProfessionalIds)
-        !== JSON.stringify(facilitySections.healthcareProfessionalIds)
-        || facilitySections.healthProfessionalsRelations.length
-
-    return areThereUnsavedChanges
+    const facilitySections = facilitiesStore.facilitySectionFields
+
+    return facilityBeforeChange.nameEn !== facilitySections.nameEn
```

Comment:
Can we pull back out the variable here? I think it's helpful to explain what this comparison is doing

Reply:
It's a best practice to have comparators as a variable